src/sunlinsol/sptfqmr/sunlinsol_sptfqmr.c

```diff
@@ -61,18 +62,15 @@ SUNLinearSolver SUNLinSol_SPTFQMR(N_Vector y, int pretype, int maxl,
   if (maxl <= 0) { maxl = SUNSPTFQMR_MAXL_DEFAULT; }
 
   /* check that the supplied N_Vector supports all requisite operations */
-  if ((y->ops->nvclone == NULL) || (y->ops->nvdestroy == NULL) ||
-      (y->ops->nvlinearsum == NULL) || (y->ops->nvconst == NULL) ||
-      (y->ops->nvprod == NULL) || (y->ops->nvdiv == NULL) ||
-      (y->ops->nvscale == NULL) || (y->ops->nvdotprod == NULL))
-  {
-    return (NULL);
-  }
+  SUNAssertNull((y->ops->nvclone) && (y->ops->nvdestroy) &&
+                  (y->ops->nvlinearsum) && (y->ops->nvconst) && (y->ops->nvprod) &&
+                  (y->ops->nvdiv) && (y->ops->nvscale) && (y->ops->nvdotprod),
+                SUN_ERR_ARG_INCOMPATIBLE);
```

Comment:
I created `SUN_ERR_ARG_INCOMPATIBLE` precisely for this situation. Assertions on numerical valued inputs usually map better to `OUTOFRANGE` since typically we are either asking for a value to be in some range explicitly or asking for an enum value which implicitly has a range.